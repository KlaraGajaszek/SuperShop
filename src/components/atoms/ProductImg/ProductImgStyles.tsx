import styled from 'styled-components';

export const ProductImage = styled.img`
  height: 100%;
  display:block;
width: 100%;
  border-radius: 8px;
  object-fit;
`;
