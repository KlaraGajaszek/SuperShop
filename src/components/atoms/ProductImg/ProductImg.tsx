import React from 'react';

import { ProductImage } from './ProductImgStyles';
import { PropsProductImage } from '../../../models/Product';

class ProductImg extends React.Component<PropsProductImage> {
  render() {
    const { imageSrc } = this.props;
    return (
      <>
        <ProductImage
          src={imageSrc}
          alt="Product Photo"
          data-testid="productImg"
        />
      </>
    );
  }
}

export default ProductImg;
