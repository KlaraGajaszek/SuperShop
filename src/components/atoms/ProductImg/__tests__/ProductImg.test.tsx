import React from 'react';

import { render, screen } from '@testing-library/react';

import ProductImg from '../ProductImg';

describe('ProductImg', () => {
  it('renders without crash', () => {
    render(<ProductImg data-testid="productImg" />);

    const img = screen.getByTestId('productImg');
    expect(img).toBeInTheDocument();
    expect(img).toMatchSnapshot();
  });

  //how I can test that good image is rendering with src

  it('render image with src prop', () => {
    render(
      <ProductImg
        data-testid="productImg"
        src="https://picsum.photos/1000/30h0"
      />
    );
    const img = screen.getByTestId('productImg');
    expect(img).toBeInTheDocument();
  });
});
