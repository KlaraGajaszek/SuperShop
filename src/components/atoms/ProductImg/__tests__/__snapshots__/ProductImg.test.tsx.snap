// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImg renders without crash 1`] = `
<img
  alt="Product Photo"
  class="sc-bdnylx lohDXh"
  data-testid="productImg"
/>
`;
